### PR TITLE
OBPIH-4932 Unable to delete request as superuser

### PR DESCRIPTION
**Pull request by kchelstowski**
I'm wondering if it's somehow possible to "inject" `def userService` like we do everywhere, so just using `def userService` at the top of file?
I had hard time to find how to use the `UserService` here, because using method like `UserService.someMethod()` didn't want to work.
@awalkowiak ?

### CODE REVIEW

**Comment by awalkowiak** [comment]
@kchelstowski IIRC you have to use `ApplicationHolder` here.

**Comment by kchelstowski** [comment]
@awalkowiak Ok, so that's what I've actually done

**Comment by drodzewicz** [comment]
is `package-lock.json` neccesary fro this PR ?

**Review by drodzewicz — CHANGES_REQUESTED** [review]
is `packeg-lock.json` necessary for this PR ?

**Comment by kchelstowski** [comment]
@drodzewicz I've just had to do `npm install` when changing to branch from `develop`'s base after a long time

File: src/groovy/org/pih/warehouse/api/StockMovement.groovy [hunk 1]
```diff
@@ -230,8 +232,8 @@ class StockMovement {
         if (electronicType) {
             User user = AuthService.currentUser.get()
             def accessRule = ConfigHelper.findAccessRule("stockRequest", "remove")
-            def userRoles = user.getEffectiveRoles(currentLocation)
-            if (!userRoles.any { Role role -> role.roleType == accessRule?.accessRules?.minimumRequiredRole }) {
+            def userService = ApplicationHolder.getApplication().getMainContext().getBean("userService")
```
Comment (jmiranda) [inline src/groovy/org/pih/warehouse/api/StockMovement.groovy:235]:
```
ApplicationHolder.application.mainContext.getBean("userService")
```

Comment (jmiranda) [reply src/groovy/org/pih/warehouse/api/StockMovement.groovy:235]: This doesn't feel right so we should add this to our tech debt pile to figure out a better way to centralize this type of custom authorization logic.

**Review by jmiranda — CHANGES_REQUESTED** [review]

File: src/groovy/org/pih/warehouse/api/StockMovement.groovy (hunk 1 above)
Comment (kchelstowski) [reply src/groovy/org/pih/warehouse/api/StockMovement.groovy:235]: @jmiranda what would you suggest at this point? I believe this one has to go to release and we already do have similar things implemented. Should we consider pushing it further and then refactor all this stuff after the tech debt or shouldn't we merge this one?

Comment (jmiranda) [reply src/groovy/org/pih/warehouse/api/StockMovement.groovy:235]: @kchelstowski Yeah I meant we should move forward with this but make sure we add this to our tech debt to discuss it further.

Comment (jmiranda) [reply src/groovy/org/pih/warehouse/api/StockMovement.groovy:235]: @kchelstowski "This doesn't feel right" was in reference to the use of the service in the StockMovement class, not my recommended change. I'd still like you to make the change suggested above.

**Review by jmiranda — APPROVED** [review]

**Pull request merged**